Java/README.md

```diff
@@ -0,0 +1,169 @@
+
+```text
+mvn package -DexcludedGroups=functional
+```
+
+## Build Command-line (CLI) usage
+
+For some of the algorithms included in this package, there are CLI applications that can
+be used for experiments. These applications use `String::split` to read
+delimited data, and as such are **not intended for production use**. Instead,
+use these applications as example code and as a way to learn about the
+algorithms and their hyperparameters.
+
+After building the project (described in the previous section), you can invoke an example CLI application by adding the
+core jar file to your classpath. For example:
+
+```text
+% java -cp core/target/randomcutforest-core-1.0.jar com.amazon.randomcutforest.runner.AnomalyScoreRunner --help
+Usage: java -cp RandomCutForest-1.0-super.jar com.amazon.randomcutforest.runner.AnomalyScoreRunner [options] < input_file > output_file
+
+Compute scalar anomaly scores from the input rows and append them to the output rows.
+
+Options:
+        --delimiter, -d: The character or string used as a field delimiter. (default: ,)
+        --header-row: Set to 'true' if the data contains a header row. (default: false)
+        --number-of-trees, -n: Number of trees to use in the forest. (default: 100)
+        --random-seed: Random seed to use in the Random Cut Forest (default: 42)
+        --sample-size, -s: Number of points to keep in sample for each tree. (default: 256)
+        --shingle-cyclic, -c: Set to 'true' to use cyclic shingles instead of linear shingles. (default: false)
+        --shingle-size, -g: Shingle size to use. (default: 1)
+        --window-size, -w: Window size of the sample or 0 for no window. (default: 0)
+
+        --help, -h: Print this help message and exit.
+```
+
+## Testing
+
+The core library test suite is divided into unit tests and "functional" tests. By "functional", we mean tests that 
+verify the expected behavior of the algorithms defined in the package. For example, a functional test for the anomaly 
+detection algorithm will first train a forest on a pre-defined distribution and then verify that the forest assigns a 
+high anomaly score to anomalous points (where "anomalous" is with respect to the specified distribution). Functional 
+tests are indicated both in the test class name (e.g., `RandomCutForestFunctionalTest`) and in a `@Tag` annotation on 
+the test class.
+
+The full test suite including functional tests currently takes over 10 minutes to complete. If you are contributing to
+this package, we recommend excluding the functional tests while actively developing, and only running the full test
+suite before creating a pull request. Functional tests can be excluded from Maven build targets by passing
+`-DexcludedGroups=functional` at the command line. For example:
+
+```text
+% mvn test -DexcludedGroups=functional
+```
+
+We currently have 90% line coverage with the full test suite, and 80% line coverage when running the unit tests only
```

Comment:
We should change this to reflect coverage in the core library, which is currently at 79% according to jacoco.